articles/building-an-appointment-scheduling-telegram-bot/index.md

```diff
@@ -0,0 +1,514 @@
+                "completed": False,
+
+                "date_due": message.split(",")[1]
+
+            }
+
+        }))
+
+        client.query(q.update(q.ref(q.collection("Users"), user["ref"].id()), {"data": {"last_command": ""}}))
+
+        context.bot.send_message(chat_id=chat_id, text="Successfully added appointment event 👍")
+
+```
+
+In the code above, you defined a method called `echo` where you passed `update` and `context` as parameters. In the function, you saved the user’s message and `chat_id`. You then made use of the `chat_id` to query the FQL client using the `users_index` index to check the user data that matches the `chat_id`.
+
+Next, you checked if the `last_command` in the data retrieved is an `add_appointment` command. If it is, you proceed to query the FQL client to create an appointment using the Fauna `create` method in the `Appointments` collection with the information provided in the user’s message.
+
+Finally, you need to create a command handler for the `add_appointment` command and a message handler for the `echo` method. The message handler filters the message and triggers the `echo` method when a message the user enters a message( a message is any text that does not start with `/`).
+
+> Note: The echo method only works when the last command was an `add_appointment` command.
+
+```python
+
+dispatcher.add_handler(CommandHandler("add_appointment", add_appointment))
+
+dispatcher.add_handler(MessageHandler(Filters.text, echo))
+
+```
+
+![add_appointment](/engineering-education/building-an-appointment-scheduling-telegram-bot/add_appointment.png)
+
+#### Listing Appointments
+
+To list all the appointments saved for a user, copy and paste the code below in your python file.
+
+```python
+
+def list_appointments(update, context):
+
+   chat_id = update.effective_chat.id
+
+
+   event_message = ""
+
+   events = client.query(q.paginate(q.match(q.index("appointment_index"), chat_id)))
+
+   for i in events["data"]:
+
+       event = client.query(q.get(q.ref(q.collection("Appointments"), i.id())))
+
+       if event["data"]["completed"]:
+
+           event_status = "Completed"
+
+       else:
+
+           event_status = "Not Completed"
+
+       event_message += "{}\nStatus:{} \nDate Due: {}\nUpdate Link: /update_{}\nDelete Link: /delete_{}\n\n".format(event["data"]["event"], event_status, event["data"]["date_due"], i.id(), i.id())
+
+   if event_message == "":
+
+       event_message = "You dont hava any appointments saved, type /add_appointment to schedule one now 😇"
+
+   context.bot.send_message(chat_id=chat_id, text=event_message)
+
+```
+
+In the code above, you defined a method called `list_appointments` where you passed `update` and `context` as parameters. In the function, you saved the `chat_id` of the user and created an empty variable called `event_message`. You then made use of the `chat_id` to query the FQL client using the `appointments_index` index to check the user data that matches the `chat_id`.
+
```

Comment:
You added this part "In the code above, you defined a method called `list_appointments` where you passed `update` and `context` as parameters" after every part of code.
Please go through the entire article and delete it. (leave the first one)